### PR TITLE
Add XML syntax for configuring SSL

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

**Comment by bjorncs** [comment]
@gjoranv Ready for review

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/ThrowingSslContextFactoryProvider.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.jdisc.http.ssl;
+
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * A dummy implementation of {@link SslContextFactoryProvider} to be injected into non-ssl connectors
```
Comment (gjoranv) [inline jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/ThrowingSslContextFactoryProvider.java]: I suggest to copy this comment into the dummy in the model as well.

**Review by gjoranv — CHANGES_REQUESTED** [review]

**Comment by bjorncs** [comment]
@gjoranv PTAL

File: config-model/src/test/java/com/yahoo/vespa/model/container/xml/JettyContainerModelBuilderTest.java [hunk 2]
```diff
@@ -229,6 +234,87 @@ public void ssl_keystore_and_truststore_configurator_can_be_overriden() throws I
         }
     }
 
+    @Test
+    public void verify_default_ssl_configuration() {
+        Element clusterElem = DomBuilderTest.parse(
```
[the remark this answers is not in the document]
Comment (tokle) [reply config-model/src/test/java/com/yahoo/vespa/model/container/xml/JettyContainerModelBuilderTest.java]: Might look a bit better, but breaks possibility to use intellij "Inject language" feature.

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java [hunk 3]
```diff
@@ -58,20 +62,61 @@ public SslContextFactory getInstance(String containerId, int port) {
...
-        factory.setProtocol(sslConfig.protocol());
+        // Check if using new ssl syntax from services.xml
+        if (!sslConfig.privateKeyFile().isEmpty()) {
+            factory.setKeyStore(createKeystore(sslConfig));
+            if (!sslConfig.caCertificateFile().isEmpty()) {
+                factory.setTrustStore(createTruststore(sslConfig));
+            }
+            factory.setProtocol("TLS");
+        } else {
+            // TODO Remove SslKeyStoreConfigurator / SslTrustStoreConfigurator
```
Comment (gjoranv) [inline jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java]: TODO Vespa 7?

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java [hunk 4]
```diff
@@ -58,20 +62,61 @@ public SslContextFactory getInstance(String containerId, int port) {
...
+            sslKeyStoreConfigurator.configure(new DefaultSslKeyStoreContext(factory));
+            sslTrustStoreConfigurator.configure(new DefaultSslTrustStoreContext(factory));
+
+            // TODO Remove support for deprecated ssl connector config
```
Comment (gjoranv) [inline jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java]: TODO Vespa 7?

Comment (bjorncs) [reply jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java]: TODO later this sprint. This was never a public api :)

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java (hunk 3 above)
Comment (bjorncs) [reply jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java]: TODO later this sprint. This was never a public api :)

File: jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java (hunk 4 above)
Comment (gjoranv) [reply jdisc_http_service/src/main/java/com/yahoo/jdisc/http/ssl/DefaultSslContextFactoryProvider.java]: Add "Vespa 7" to the comment

**Review by gjoranv — APPROVED** [review]

**Comment by bjorncs** [comment]
Rebased with master

**Pull request merged**